auto_dev/commands/scaffold.py

```diff
@@ -295,69 +300,111 @@ def handler(ctx, spec_file, public_id, new_skill, auto_confirm) -> int:
     help="The type of behaviour to generate.",
     default=BehaviourTypes.metrics,
 )
+@click.argument("skill_public_id", type=PublicId.from_str, required=False)
 @click.pass_context
 def behaviour(
     ctx,
-    spec_file,
-    behaviour_type,
-    auto_confirm,
-    target_speech_acts,
+    spec_file: str,
+    behaviour_type: str,
+    target_speech_acts: str | None,
+    auto_confirm: bool,
+    skill_public_id: PublicId | None,
 ) -> None:
-    """Generate an AEA handler from an OpenAPI 3 specification.
+    """Scaffold a behaviour from a specification.
 
-    Example:
-    -------
-    ```
-    adev scaffold behaviour openapi.yaml --behaviour-type metrics
-    ```
+    Required arguments:
+        spec_file: Path to specification file (OpenAPI for metrics, FSM for simple_fsm)
+        behaviour_type: Type of behaviour to scaffold (metrics or simple_fsm)
 
+    Optional arguments:
+        target_speech_acts: Comma separated list of speech acts to scaffold
+        auto_confirm: Auto confirm all actions
+        skill_public_id: Optional public ID of the skill. If not provided, assumes current directory is a skill.
     """
     logger = ctx.obj["LOGGER"]
     verbose = ctx.obj["VERBOSE"]
 
+    if not Path(DEFAULT_AEA_CONFIG_FILE).exists():
+        msg = f"No {DEFAULT_AEA_CONFIG_FILE} found in current directory, please run `adev create` to create an Agent."
+        raise ValueError(msg)
+
+    if not Path(spec_file).exists():
+        msg = f"Specified spec '{spec_file}' does not exist."
+        raise click.ClickException(msg)
+
     scaffolder = BehaviourScaffolder(
         spec_file,
         behaviour_type=BehaviourTypes[behaviour_type],
         logger=logger,
         verbose=verbose,
         auto_confirm=auto_confirm,
     )
-    scaffolder.scaffold(
-        target_speech_acts=target_speech_acts,
-    )
+    scaffolder.scaffold(target_speech_acts=target_speech_acts)
 
+    generated_behavior = scaffolder.generated_behavior
+    if not generated_behavior:
+        msg = "No behavior generated from specification"
+        raise ValueError(msg)
 
-@scaffold.command()
-@click.argument("spec_file", type=click.Path(exists=True), required=True)
-@click.option("-tsa", "--target-speech-acts", default=None, help="Comma separated list of speech acts to scaffold.")
-@click.option("--auto-confirm", is_flag=True, default=False, help="Auto confirm all actions")
-@click.option(
-    "--handler_type",
-    type=click.Choice([HandlerTypes.simple]),
-    required=True,
-    help="The type of behaviour to generate.",
-    default=HandlerTypes.simple,
-)
-@click.pass_context
-def handlers(ctx, spec_file, handler_type, auto_confirm, target_speech_acts) -> None:
-    """Generate an AEA handler from an OpenAPI 3 specification.
+    if skill_public_id:
+        with change_dir(f"skills/{skill_public_id.name}"):
```

Comment:
remove as assumes it is ejected and that we are in an agent. lots of assumptions that are checked in the agent runer for example!